blocksync/buffer.go

```diff
@@ -78,11 +73,8 @@ func (b *blockBuffer) Flush(blk *blockchain.Block) (bool, bCheckinResult) {
 		}
 		if err := commitBlock(b.bc, b.ap, b.blocks[syncHeight]); err == nil {
 			syncedHeight = syncHeight
-			if !b.blocks[syncedHeight].IsDummyBlock() {
-				b.confirmedHeight = syncedHeight
-			}
-			l.Info().Uint64("syncedHeight", syncedHeight).Bool("dummy", b.blocks[syncedHeight].IsDummyBlock()).
-				Msg("Successfully committed block.")
+			b.confirmedHeight = syncedHeight
```

Comment:
if there is no dummy block, we can merge `confirmedHeight` with `startHeight`. `startHeight` is just `confirmedHeight` + 1